### PR TITLE
Ardougne region and Varrock overhaul

### PR DESCRIPTION
**Pull request by SirFancyBacon**
Fixes a lot of nagging issues and adds textures to most objects within East and West Ardougne with the exception of wooden objects; Because of the sheer amount of objects that are shared between Varrock and Arougne i have also improved various small textures around Varrock.

Biggest changes in this PR are most doors have textures in the overworld now, Metal fences have a small amount of specular highlights.

Strongly encourage testers to load up the PR and run around Varrock and Ardougne for out of place textures.

Metallic armor, furnaces, Wall torches:
Screenshot from 2022-10-31 18-56-10
Screenshot from 2022-10-31 19-03-45
Screenshot from 2022-10-31 19-56-09
Screenshot from 2022-10-31 20-15-25

Improved textures around Ardougne:

Screenshot from 2022-10-30 21-15-33
Screenshot from 2022-10-30 21-23-10
Screenshot from 2022-10-30 21-07-26
Screenshot from 2022-10-30 21-09-00
Screenshot from 2022-10-30 20-43-41
Screenshot from 2022-10-30 20-47-47

### CODE REVIEW

**Comment by SirFancyBacon** [comment]
Overhaul of wooden objects on hold until #71 is merged

**Comment by SirFancyBacon** [comment]
Also included are the pillared walls; Just forgot to nab new screens

**Comment by SirFancyBacon** [comment]
This PR is not ready to be merged as i expanded in e27878c to include the new gravel texture i was working on as well as a ton of objects missing textures around Varrock.

I encourage anyone to test these to see how you like it; will post screens later

**Comment by SirFancyBacon** [comment]
This is ready to be reviewed

**Review by aHooder — CHANGES_REQUESTED** [review]
A lot of this stuff is ready to merge I'd say, good job :+1:

I haven't been able to look through most of the model override changes, as those are probably scattered all over the place, but they seem good from the JSON.

Texture additions I'd try to keep to a minimum, since we have that hard limit to consider, but one metallic texture addition should be fine imo. I think metal should ideally use roughness maps instead of diffuse color textures, or maybe a mixture of both for things like rust spots, but that'll have to wait for more proper support, so this is good in the meantime.

In future PRs, it would probably be nice to separate some of these things into multiple PRs, like metallic stuff in one, path fixes in another, misc object fixes in a third and tile texture changes in a fourth. That way some small issues won't hold back all of the changes :sweat_smile:

File: src/main/java/rs117/hd/data/environments/Area.java
Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/environments/Area.java:323]: Both of these are for under the bridge, Thought it looked nicer personally. line 321 should have the bridge itself so these two lines can be removed

File: src/main/java/rs117/hd/data/materials/Material.java
Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/materials/Material.java:228]: I changed the gravel texture with one i think is far superior; If we don't like it we can scrap them.
Scroll to the bottom to see:
https://github.com/117HD/RLHD/pull/102/commits/98c4b036fa144364e9cbca9eec6d8f79fa532ba6

File: src/main/java/rs117/hd/data/materials/Material.java
Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/materials/Material.java:363]: 256x256 is a bit low, the contrast just isn't enough so it looks like no texture at a distant and a fake texture up close.
But i am happy to try a new texture, Perhaps i just need less markings to "brighten" the texture

**Comment by SirFancyBacon** [comment]
> A lot of this stuff is ready to merge I'd say, good job 👍

Thank you :)

> Texture additions I'd try to keep to a minimum, since we have that hard limit to consider, but one metallic texture addition should be fine imo. I think metal should ideally use roughness maps instead of diffuse color textures, or maybe a mixture of both for things like rust spots, but that'll have to wait for more proper support, so this is good in the meantime.

Absolutely, I always try textures that are already existing before i make a new one. But if ever you feel it is not good to have or can suggest an alternative i am always open! your UVs help a ton with walls and floors.

> In future PRs, it would probably be nice to separate some of these things into multiple PRs, like metallic stuff in one, path fixes in another, misc object fixes in a third and tile texture changes in a fourth. That way some small issues won't hold back all of the changes 😅

I figured a big PR would be better since you have limited time; I will break them apart in the future (i'll need to remake my Dorgesh-kaan overhual now lol)

File: src/main/java/rs117/hd/data/environments/Area.java
Comment (aHooder) [reply src/main/java/rs117/hd/data/environments/Area.java]: Ah, I see what you mean. I guess it's all pretty subjective :slightly_smiling_face: I'll add a commit to remove those lines and properly rename that path blend fix where it's referenced as well.

**Comment by aHooder** [comment]
> > In future PRs, it would probably be nice to separate some of these things into multiple PRs, like metallic stuff in one, path fixes in another, misc object fixes in a third and tile texture changes in a fourth. That way some small issues won't hold back all of the changes sweat_smile
> 
> I figured a big PR would be better since you have limited time; I will break them apart in the future (i'll need to remake my Dorgesh-kaan overhual now lol)

Haha, I see. I'd say smaller PRs are easier, even if there's a lot of them. You don't have to be super accurate when separating things out, and you don't really have to bother if you've already got a massive one, but might be nice to keep in mind for future stuff.

File: src/main/java/rs117/hd/data/materials/Material.java
Comment (aHooder) [reply src/main/java/rs117/hd/data/materials/Material.java:228]: Looking at this side by side, I think I might have to agree that it's superior. This might just be some innate resistance to change on my part :sweat_smile: It certainly feels more natural, but maybe less characteristic?

image
image
image
image

The main part lacking I'd say is detail up close, which might be possible to achieve by tuning specularity and adding a normal map. That's definitely worth a shot I'd say.

**Comment by aHooder** [comment]
One thing I forgot to mention is that you should specify where the textures come from, even if they're public domain, so we can verify that they can be used and distributed by RuneLIte.

File: src/main/java/rs117/hd/data/materials/Material.java
Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/materials/Material.java:350]: This is exactly what you are asking for, the one Metallic texture but with specular set to a highgloss. No extra date needed :)

Comment (aHooder) [reply src/main/java/rs117/hd/data/materials/Material.java:352]: Yeah, that's all good, but I meant that `METALLIC_2` looks like the same texture as `METALLIC_1`, just with altered brightness, yet it has its own corresponding texture file. I was wondering if it might make sense to add support for this brightness change with a new material property instead? Maybe just a brightness or opacity factor could make sense?

Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/materials/Material.java:352]: Oh! i am following you; Absolutely if you could git me a way to tweak brightness that would be a HUGE improvement to so many textures

**Comment by SirFancyBacon** [comment]
image
new CCO compatible texture for gravel

**Comment by SirFancyBacon** [comment]
image
zoomed out view

**Comment by SirFancyBacon** [comment]
Updated wattle:
image

**Comment by SirFancyBacon** [comment]
The last commit fixes the metal file

File: src/main/java/rs117/hd/data/materials/Material.java
Comment (aHooder) [reply src/main/java/rs117/hd/data/materials/Material.java:228]: I'm fairly happy with how the gravel looks at this point, but it might be worth considering keeping the old texture available as a toggle, since this changes a core texture that's seen in a lot of places.
Updated screenshots:
image
image

Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/materials/Material.java:228]: How would i go about making a toggle for said texture?

Comment (aHooder) [reply src/main/java/rs117/hd/data/materials/Material.java:228]: Added a toggle for this in https://github.com/117HD/RLHD/pull/102/commits/aef1e617337144022f173d7c4db5331f744b94e0

**Comment by aHooder** [comment]
I think this is good to go now. Anything else you want to get in?

**Comment by SirFancyBacon** [comment]
Nope, this should be it for this one, I broke up my new ones into small ones :D

**Pull request merged**
